src/test/java/com/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,175 @@
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(UserController.class)
+
+
+class UserControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    @MockBean
+    private User user;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setUp() {
+        user = new User("jhon_doe", "JhonDoe", LocalDate.parse("1990-01-01"));
+    }
+
+    @Test
+    public void givenUsers_whenGetUsers_thenReturnsJsonArray() throws Exception {
+
+        List<User> allUsers = Collections.singletonList(user);
+
+        given(userRepository.findAll()).willReturn(allUsers);
+
+        mockMvc.perform(get(URI.create("/api/users"))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", hasSize(1)))
+            .andExpect(jsonPath("$[0].name", is(user.getName())));
+    }
+
+    @Test
+    public void givenUsers_whenGetUser_thenReturnsJson() throws Exception {
+
+        given(userRepository.findById(user.getId())).willReturn(ofNullable(user));
+
+        mockMvc.perform(get(URI.create(String.format("/api/users/%d", user.getId())))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.name", is(user.getName())));
+    }
+
+    @Test
+    public void givenValidUser_whenPostUser_thenReturnsJson() throws Exception {
+
+        given(userRepository.save(user)).willReturn(user);
+
+        mockMvc.perform(post(URI.create("/api/users"))
+            .content(objectMapper.writeValueAsString(user))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isCreated());
+    }
+
+    @Test
+    public void givenInvalidUser_whenPostUser_thenReturnsBadRequest() throws Exception {
+        user = new User();
+        user.setName("Jane Doe");
+        user.setUsername("jane_doe");
+        given(userRepository.save(user)).willReturn(user);
+
+        mockMvc.perform(post(URI.create("/api/users"))
+            .content(objectMapper.writeValueAsString(user))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void givenValidUser_whenDeletesUser_thenReturnsNoContent() throws Exception {
```

Comment:
Same here, it would be nice to have a test to check that the "not found" exception is thrown when necessary